### PR TITLE
enable regex token replacement

### PR DESCRIPTION
**Pull request by sbma44**
This functionality is already somewhat in place due to the nature of current token processing -- tokens containing regex characters could already behave in unexpected ways. This PR alters that codepath to use the XRegExp module.

- implement named group capture, making existing boundary checking safer and allowing for their use in replacement patterns
- check for numbered groups in the input and throw errors when they are encountered (carmen introduces new groups, making the ordering dangerous--we would have to do some fancy rewrite stuff which is best avoided in favor of named groups)

### CODE REVIEW

**Comment by sbma44** [comment]
@ingalls ok to merge?

**Comment by ingalls** [comment]
^^ Mind merging in master first and making this green?

**Comment by sbma44** [comment]
ok, green! give it a once-over whenever you have a sec, please

File: package.json [hunk 1]
```diff
@@ -23,7 +24,8 @@
     "turf-linestring": "1.0.x",
     "turf-point": "2.0.x",
     "turf-point-on-surface": "1.0.x",
-    "unidecode": "0.1.3"
+    "unidecode": "0.1.3",
+    "xregexp": "^2.0.0"
```
Comment (ingalls) [inline package.json]: I'd lock this to a specific version or at least to the patch level, the caret will match any minor version

**Comment by ingalls** [comment]
Looks great @sbma44

**Pull request merged**